player/lua/ytdl_hook.lua

```diff
@@ -1006,6 +1006,16 @@ local function run_ytdl_hook(url)
     if result.status ~= 0 or json == "" then
         json = nil
     elseif json then
+        -- trim any preceding text before JSON
+        local start = string.find(json, "{", 1, true)
+        if start ~= 1 then
+            if start == nil then
```

Comment:
This block can be placed above with less nesting. Also you can use `json:find()` and `json:sub()`.

Reply:
Technically yes but it's not logically consistent, we only want to process the JSON if the command was successful.

Regarding the `:` calling syntax, my Lua is a bit rusty so I'll have to look up the docs again to how it works.

Reply:
Well I was mainly referring to moving the `start == nil` block up. I didn't know if skipping the `sub` call when `start` is 1 makes a meaningful difference since this isn't a loop but you can still check `if start > 1`.

Reply:
That makes sense, I will modify the code.

Reply:
That makes sense, I will modify the code.

Reply:
That is nicer, thanks.